.github/workflows/ofrak-type-ci.yml

```diff
@@ -0,0 +1,619 @@
+name: OFRAK Type, IO, Patch Maker, and Core CI
+
+on:
+  pull_request:
+    paths:
+      - 'ofrak_type/**'
+      - 'ofrak_io/**'
+      - 'ofrak_patch_maker/**'
+      - 'pytest_ofrak/**'
+      - 'ofrak_core/**'
+      - 'frontend/**'
+      - '.github/workflows/ofrak-type-ci.yml'
+  push:
+    branches:
+      - feature/windows_tests
+    #      - master
+#    paths:
+#      - 'ofrak_type/**'
+#      - 'ofrak_io/**'
+#      - '.github/workflows/ofrak-type-ci.yml'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash  # NOTE: On Windows, this uses Git Bash which provides Unix compatibility (for example for ofrak-io's StreamCapture
+
+jobs:
+  build-ofrak-type-sdist:
+    name: build ofrak-type sdist
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install build tools
+        run: |
+          python -m pip install -r pip-requirements.txt
+          pip install build
+
+      - name: Build OFRAK Type source distribution
+        run: |
+          cd ofrak_type
+          python -m build --sdist --outdir ../dist-type
+
+      - name: List built packages
+        run: |
+          echo "Built OFRAK Type packages:"
+          ls -la dist-type/
+
+      - name: Upload source distribution
+        uses: actions/upload-artifact@v4
+        with:
+          name: ofrak-type-sdist
+          path: dist-type/
+          retention-days: 30
```

Comment:
This is essentially a duplicate of the sdist build workflow in the other file. Maybe it should run on a (1D) matrix over `ofrak-type`, `ofrak-io`, OFRAK core, etc.